src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/LocalizedValidationMetadataProvider.cs

```diff
@@ -0,0 +1,45 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
+using Microsoft.Extensions.Localization;
+
+namespace OrchardCore.Localization.DataAnnotations
+{
+    public class LocalizedValidationMetadataProvider : IValidationMetadataProvider
+    {
+        private readonly IStringLocalizer _stringLocalizer;
+
+        public LocalizedValidationMetadataProvider(IStringLocalizer stringLocalizer)
+        {
+            _stringLocalizer = stringLocalizer;
+        }
+
+        /// <remarks>This will localize the default data annotations error message if it is exist, otherwise will try to look for a parameterized version.</remarks>
+        /// <example>
+        /// A property named 'UserName' that decorated with <see cref="RequiredAttribute"/> will be localized using
+        /// "The {0} field is required." and "The UserName field is required." error messages.
+        /// </example>
+        public void CreateValidationMetadata(ValidationMetadataProviderContext context)
+        {
+            foreach (var metadata in context.ValidationMetadata.ValidatorMetadata)
+            {
+                var attribute = metadata as ValidationAttribute;
+                var argument = context.Key.Name;
+                var errorMessageString = attribute != null && attribute.ErrorMessage == null && attribute.ErrorMessageResourceName == null
+                    ? attribute.FormatErrorMessage(argument)
+                    : attribute.ErrorMessage;
+
+                // Localize the error message without params
+                var localizedString = _stringLocalizer[errorMessageString];
+
+                if (localizedString == errorMessageString)
+                {
+                    // Localize the error message with params
+                    errorMessageString = errorMessageString.Replace(argument, "{0}");
+                    localizedString = _stringLocalizer[errorMessageString, argument];
```

Comment:
I think this won't take the `Display` attribute into account when parameterizing the error message.

Since the `argument ` (`context.Key.Name`) is the property name here, the property name will display in the error message, even if the view model property has a `Display `attribute and there is a translation in the PO file.

In other words, if I add a `Display` attribute to a view model property like this

```
[Display(Name = "Default Sender")]
public string DefaultSender { get; set; }
```
and have entries in my en-US.po file like this

```
msgid "The {0} field is required."
msgstr "Translated message: {0} is required."

msgid "Default Sender"
msgstr "Translated Default Sender"
```

the localized error message will be "Translated message: DefaultSender is required."

However, I would expect the localized error message to take into account the localized display name, so it would be "Translated message: Translated Default Sender is required."

I think we can leave the error message unparameterized here.

```suggestion
                    localizedString = _stringLocalizer[errorMessageString];
```

It looks like [DataAnnotationsMetadataProvider.CreateDisplayMetadata](https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.DataAnnotations/DataAnnotationsMetadataProvider.cs#L151) ultimately parameterizes the error message and takes into account the localized display name.

![image](https://user-images.githubusercontent.com/15861981/118012449-11dd3580-b31f-11eb-9626-7e8a808f7fc8.png)

If we make the suggested change above and add `builder.AddDataAnnotationsLocalization()` back into OrchardCore.Mvc.Core.Startup.cs (as suggested in a separate comment in this review), the localized error message correctly includes the localized display name as a parameter.

![image](https://user-images.githubusercontent.com/15861981/118016034-e9efd100-b322-11eb-9600-7df59e9983c2.png)